src/server/Services/GatewayService.cs

```diff
@@ -28,6 +40,12 @@ public sealed class GatewayService(DataLayerProxy dataLayer,
 
     private static string GetAssemblyVersion() => Assembly.GetExecutingAssembly().GetName().Version?.ToString() ?? "0.0.0.0";
 
+    public Task<bool> InvalidateStore(string storeId)
+    {
+        _fileCache.InvalidateStore(storeId);
```

Comment:
@dkackman Filecache is invalidating correctly, however im really stuggling on how to invalidate memory cache since its keys dont seem to be iterable. I want to invalidate all keys that start with the storeId in the cacheKey.

Reply:
will look but in a few hours

Reply:
with this simple IMemoryCache that isn't possible. You can remove a given item by key but not iterate the contents (at least based on what I can tell). We'd have change the caching mechanism or keep a parallel list of cache keys.

Reply:
i looked into a parrallel list of cache keys and it just seemed super convoluted.

Here is an idea, the filecache is named after the cache key, maybe when we iterate through the filecache we delete the memory cache with the same key